### PR TITLE
[20565] Added Annotation dependencies

### PR DESCRIPTION
**Pull request by adriancampo**
This PR adds the Annotations when solving dependencies.

## Contributor Checklist
- [X] Commit messages follow the project guidelines. 
- [X] The code follows the style guidelines of this project. 
- [N/A] Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- [N/A] Any new/modified methods have been properly documented using Doxygen. 
- [N/A] Changes are ABI compatible. 
- [N/A] Changes are API compatible. 
- [N/A] New feature has been added to the `versions.md` file (if applicable).
- [N/A] New feature has been documented/Current behavior is correctly described in the documentation. 
    
- [N/A] Applicable backports have been included in the description.

## Reviewer Checklist
- [ ] The PR has a milestone assigned.
- [ ] The title and description correctly express the PR's purpose.
- [ ] Check contributor checklist is correct.
- [ ] Check CI results: changes do not issue any warning.
- [ ] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

File: src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.cpp [hunk 1]
```diff
@@ -518,30 +518,140 @@ ReturnCode_t TypeObjectRegistry::get_dependencies_from_type_object(
...
+                    break;
                 case TK_ENUM:
+                    if (ret_code == eprosima::fastdds::dds::RETCODE_OK &&
+                            type_object.complete().enumerated_type().header().detail().ann_custom().has_value())
+                    {
+                        ret_code = get_custom_annotations_dependencies(
+                            type_object.complete().enumerated_type().header().detail().ann_custom().value(),
+                            type_dependencies);
+                    }
+                    break;
```
Comment (JLBuenoLopez-eProsima) [inline src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.cpp:646]: CompleteEnumeratedLiteral also has a CompleteMemberDetail with custom annotations

File: src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.cpp [hunk 2]
```diff
@@ -518,30 +518,140 @@ ReturnCode_t TypeObjectRegistry::get_dependencies_from_type_object(
...
                 case TK_BITMASK:
+                    if (ret_code == eprosima::fastdds::dds::RETCODE_OK &&
+                            type_object.complete().bitmask_type().header().detail().ann_custom().has_value())
+                    {
+                        ret_code = get_custom_annotations_dependencies(
+                            type_object.complete().bitmask_type().header().detail().ann_custom().value(),
+                            type_dependencies);
+                    }
```
Comment (JLBuenoLopez-eProsima) [inline src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.cpp:654]: CompleteBitflag also contains a CompleteMemberDetail with custom annotations

**Review by JLBuenoLopez-eProsima — CHANGES_REQUESTED** [review]

File: src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.cpp (hunk 2 above)
Comment (adriancampo) [reply src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.cpp:668]: Added in [e6bbcb4](https://github.com/eProsima/Fast-DDS/pull/4497/commits/e6bbcb4eef25966bbcbe8431ae140705ef498cea)

File: src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.cpp (hunk 1 above)
Comment (adriancampo) [reply src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.cpp]: Added in [e6bbcb4](https://github.com/eProsima/Fast-DDS/pull/4497/commits/e6bbcb4eef25966bbcbe8431ae140705ef498cea)

File: src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.cpp [hunk 3]
```diff
@@ -518,30 +518,140 @@ ReturnCode_t TypeObjectRegistry::get_dependencies_from_type_object(
...
+                                    break;
+                                }
+                            }
+                        }
+                    }
                     break;
                 case TK_SEQUENCE:
                     ret_code = get_sequence_array_dependencies(type_object.complete().sequence_type(),
                                     type_dependencies);
+                    //TODO Annotations are not currently supported, so their dependecies are ignored.
```
[the remark this answers is not in the document]
Comment (adriancampo) [reply src/cpp/fastdds/xtypes/type_representation/TypeObjectRegistry.cpp]: Changed in [e6bbcb4](https://github.com/eProsima/Fast-DDS/pull/4497/commits/e6bbcb4eef25966bbcbe8431ae140705ef498cea)

**Review by JLBuenoLopez-eProsima — APPROVED** [review]
LGTM with green CI

**Comment by JLBuenoLopez-eProsima** [comment]
@richiprosima please test_3 this

**Comment by JLBuenoLopez-eProsima** [comment]
Manual CI has build correctly:

* [Ubuntu](https://github.com/eProsima/Fast-DDS/actions/runs/8359705055/job/22883594611)
* [Windows](https://github.com/eProsima/Fast-DDS/actions/runs/8359707318/job/22883602729)
* [MacOS](https://github.com/eProsima/Fast-DDS/actions/runs/8359701193/job/22883584403)

**Pull request merged**